plugins/experimental/ts_lua/ts_lua_http.c

```diff
@@ -745,6 +754,77 @@ ts_lua_http_get_server_state(lua_State *L)
   return 1;
 }
 
+static int
+ts_lua_http_get_remap_from_url(lua_State *L)
+{
+  char output[TS_LUA_MAX_URL_LENGTH];
+  int output_len;
+  TSMLoc url = TS_NULL_MLOC;
+  char *str  = NULL;
+  int len;
+
+  ts_lua_http_ctx *http_ctx;
+
+  GET_HTTP_CONTEXT(http_ctx, L);
+
+  if (TSRemapFromUrlGet(http_ctx->txnp, &url) != TS_SUCCESS) {
+    lua_pushnil(L);
+    goto done;
+  }
+
+  str = TSUrlStringGet(NULL, url, &len);
+
+  output_len = snprintf(output, TS_LUA_MAX_URL_LENGTH, "%.*s", len, str);
```

Comment:
I don't know Lua APIs much, but why is it necessary to snprintf() to the output here? Since lua_pushlstring() takes a width (length) of the string, can't we just always do e.g.

```
lua_pushlstring(L, str, len);
```

?? If this a recurring pattern in the Lua plugin, and assuming lua_pushlstring() can handle the non-null terminated string, I think we ought to fix this consistently. This is a significant inefficiency  :-).

Reply:
Yeah. This is a recurring pattern in the lua plugin. You want the fix in this PR or a separate PR?